app/assets/javascripts/utils/parseCommitUrl.js

```diff
@@ -0,0 +1,21 @@
+parseCommitUrl = function(url){
+  var changes = JSON.parse(httpGet(getCommitApiUrl(url)))["files"];
+
+  var changed_files = "";
+
+  for (var i = 0; i < changes.length; i++) {
```

Comment:
Faz um cache numa variáve nesse `changes.length` e usa no `for`. 
É uma boa prática de performance em js.